mlrun/utils/notifications/notification_pusher.py

```diff
@@ -36,55 +50,170 @@ def __init__(self, runs: typing.Union[mlrun.lists.RunList, list]):
+            notification.status
+            and notification.status != NotificationStatus.PENDING.value
+        ):
+            return False
 
         # if at least one condition is met, notify
         for when_state in when_states:
             if (
-                when_state == "success"
-                and run_state == "success"
+                when_state == run_state == "completed"
                 and (not condition or ast.literal_eval(condition))
-            ) or (when_state == "failure" and run_state == "error"):
+            ) or when_state == run_state == "error":
                 return True
 
         return False
 
-    def _load_notification(self, notification_config: dict) -> NotificationBase:
-        params = notification_config.get("params", {})
+    def _load_notification(
+        self, run: mlrun.model.RunObject, notification: mlrun.model.Notification
+    ) -> NotificationBase:
+        name = notification.name
         notification_type = NotificationTypes(
-            notification_config.get("type", "console")
+            notification.kind or NotificationTypes.console
         )
-        if notification_type not in self._notifications:
+        notification_key = f"{run.metadata.uid}-{name or notification_type}"
+        params = self._load_notification_params(notification)
+        if notification_key not in self._notifications:
             self._notifications[
-                notification_type
-            ] = notification_type.get_notification()(params)
+                notification_key
+            ] = notification_type.get_notification()(name, params)
         else:
-            self._notifications[notification_type].load_notification(params)
+            self._notifications[notification_key].load_notification(params)
+
+        logger.debug(
+            "Loaded notification", notification=self._notifications[notification_key]
+        )
+        return self._notifications[notification_key]
+
+    @staticmethod
+    def _load_notification_params(notification: mlrun.model.Notification) -> dict:
+        params = notification.params or {}
+        params_secret = params.get("secret", "")
+        if not params_secret:
+            return params
+
+        k8s = mlrun.api.utils.singletons.k8s.get_k8s()
+        if not k8s:
+            raise mlrun.errors.MLRunRuntimeError(
+                "Not running in k8s environment, cannot load notification params secret"
+            )
+
+        encoded_params = k8s.load_secret(params_secret)
+        params = {}
+        if encoded_params:
+            for key, value in encoded_params.items():
+                if isinstance(value, str):
+                    value = value.encode("utf-8")
+                params[key] = base64.decodebytes(value).decode("utf-8")
+        return params
+
+    async def _send_notification(
+        self,
+        notification: NotificationBase,
+        run: mlrun.model.RunObject,
+        notification_model: mlrun.model.Notification,
```

Comment:
```suggestion
        notification_object: mlrun.model.Notification,
```
We're not getting the model here, we're getting an object that is an instance of the model.